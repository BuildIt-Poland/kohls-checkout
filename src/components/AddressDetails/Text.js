import styled from 'styled-components';

import { default as DefaultText } from '../Text';

const Text = styled(DefaultText)`
  margin: 0;
`;

export default Text;
