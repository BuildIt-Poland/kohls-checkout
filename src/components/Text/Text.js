import styled from 'styled-components';
import PropTypes from 'prop-types';

import {
  fontMedium,
  fontWeightNormal,
  fontWeightBold,
  spacingSmall,
  colorBlack,
  colorDarkGrey,
  colorTurquoiseDimmed
} from '../../styles/designTokens';

export function fontColor({ highlight, dimmed }) {
  if (highlight) {
    return colorTurquoiseDimmed;
  } else if (dimmed) {
    return colorDarkGrey;
  }
  return colorBlack;
}

const Text = styled.p`
  font-size: ${fontMedium};
  text-decoration: ${({ underline }) => (underline ? 'underline' : 'none')};
  font-weight: ${({ bold }) => (bold ? fontWeightBold : fontWeightNormal)};
  line-height: 1.4;
  margin: 0;
  margin-bottom: ${spacingSmall};
  color: ${fontColor};

  &:last-child {
    margin-bottom: 0;
  }
`;

Text.propTypes = {
  /** Highlight text */
  highlight: PropTypes.bool,
  /** Dimmed text */
  dimmed: PropTypes.bool,
  /** Underline text */
  underline: PropTypes.bool,
  /** Bold text */
  bold: PropTypes.bool
};

export default Text;
