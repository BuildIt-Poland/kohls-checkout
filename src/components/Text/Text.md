```js
<Text>Basic Text</Text>
<Text dimmed>Dimmed Text</Text>
<Text bold>Bold Text</Text>
<Text underline>Underline Text</Text>
<Text highlight>Highlight Text</Text>
<Text highlight bold>Bold hihglight Text</Text>
```
