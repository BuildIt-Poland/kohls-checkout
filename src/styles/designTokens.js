// Colors
export const colorBlack = '#000000';
export const colorDarkGrey = '#454545';
export const colorWhite = '#ffffff';
export const colorTurquoise = '#00a09f';
export const colorRed = '#e0151d';
export const colorLightRed = '#fadcdd';
export const colorLightGrey = '#b0b0b0';

// Spacing (mainly for paddings and margins)
export const spacingLarge = '3rem';
export const spacingMedium = '2rem';
export const spacingSmall = '1rem';
export const borderWidthThick = '2px';
export const borderWidthThin = '1px';

// Radius
export const bordeRadiusCircle = '50%';
export const borderRadiusMedium = '0.25rem';

// Font size
export const fontSmall = '1.2rem';
export const fontMedium = '1.8rem';
export const fontLarge = '2.4rem';

// Font weight
export const fontWeightNormal = 400;
export const fontWeightBold = 700;
export const fontWeightThin = 300;
