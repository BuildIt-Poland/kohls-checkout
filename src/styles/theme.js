const theme = {
  // Colors
  colorWhite: '#ffffff',
  colorTurquoise: '#00a09f',

  // Radius
  bordeRadiusCircle: '50%',
  borderRadiusMedium: '0.25rem'
};

export default theme;
