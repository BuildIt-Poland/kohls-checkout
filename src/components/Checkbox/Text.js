import styled from 'styled-components';

import { fontWeightNormal, spacingSmall, fontMedium, colorBlack } from '../../styles/designTokens';

const Text = styled.span`
  display: block;
  color: ${colorBlack};
  font-size: ${fontMedium};
  font-weight: ${fontWeightNormal};
  margin: ${spacingSmall};
  cursor: pointer;
`;

export default Text;
