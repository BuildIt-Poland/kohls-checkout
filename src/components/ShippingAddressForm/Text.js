import styled from 'styled-components';

import { colorDarkGrey, spacingMedium } from '../../styles/designTokens';
import TextBase from '../Text';

const Text = styled(TextBase)`
  color: ${colorDarkGrey};
  margin-bottom: ${spacingMedium};
`;

export default Text;
