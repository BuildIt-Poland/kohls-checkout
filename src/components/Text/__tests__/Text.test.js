import React from 'react';
import { create } from 'react-test-renderer';

import { colorBlack, colorDarkGrey, colorTurquoiseDimmed } from '../../../styles/designTokens';
import Text, { fontColor } from '../Text';

describe('COMPONENT - Text', () => {
  it('renders Text component', () => {
    const component = create(<Text />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('FUNC - fontColor', () => {
  it("return proper color if 'highlight' is passed", () => {
    expect(fontColor({ highlight: true })).toEqual(colorTurquoiseDimmed);
  });

  it("return proper color if 'dimmed' is passed", () => {
    expect(fontColor({ dimmed: true })).toEqual(colorDarkGrey);
  });

  it("return proper color if both 'highlight' and 'dimmed' are passed", () => {
    expect(fontColor({ highlight: true, dimmed: true })).toEqual(colorTurquoiseDimmed);
  });

  it("return proper color if 'highlight' is passed", () => {
    expect(fontColor({ highlight: false, dimmed: false })).toEqual(colorBlack);
  });
});
