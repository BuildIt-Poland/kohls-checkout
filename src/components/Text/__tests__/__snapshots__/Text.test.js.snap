// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Text renders Text component 1`] = `
.c0 {
  color: #222222;
  font-size: 1.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 400;
  font-weight: 400;
  line-height: 1.4;
  margin: 0;
  margin-bottom: 1rem;
}

.c0:last-child {
  margin-bottom: 0;
}

<p
  className="c0"
/>
`;
